### PR TITLE
357 previewsender should show a helpful message when your transport is not connected

### PR DESCRIPTION
**Pull request by alexfarrill**
## Checklist before requesting a review

- [x] I have performed a self-review of my code
- [x] If it is a core feature, I have added thorough tests.

### CODE REVIEW

File: packages/cli/src/components/PreviewSender.tsx [hunk 1]
```diff
@@ -12,19 +12,24 @@ const PreviewSender: React.FC<PreviewSenderProps> = ({
...
       const lastSent = window.localStorage.getItem("previewSenderLastSentAt");
       if (lastSent) {
         setLastSentAt(new Date(lastSent));
       }
+
+      // ensure this only happens once
+      setUsedEmailFromLocalStorage(true);
     }
-  }, [email]);
+  }, [email, usedEmailFromLocalStorage]);
```
Comment (alexfarrill) [inline packages/cli/src/components/PreviewSender.tsx:32]: There's a way to do this with less code by passing [] (https://reactjs.org/docs/hooks-effect.html), but I would have to disable the tslint for this line:  "React Hook useEffect has missing dependencies: 'email' and 'usedEmailFromLocalStorage'. Either include them or remove the dependency array."

> If you want to run an effect and clean it up only once (on mount and unmount), you can pass an empty array ([]) as a second argument. This tells React that your effect doesn’t depend on any values from props or state, so it never needs to re-run. This isn’t handled as a special case — it follows directly from how the dependencies array always works.

wdyt?

File: packages/cli/src/components/PreviewSender.tsx [hunk 2]
```diff
@@ -98,10 +113,20 @@ const PreviewSender: React.FC<PreviewSenderProps> = ({
...
+        <input
+          type="submit"
+          value="Send"
+          disabled={!email?.length}
+          className="text-sm border-none p-[9px] rounded-r-sm cursor-pointer bg-[#fff] text-black hover:bg-white"
         />
-        <input type="submit" value="Send" disabled={!email?.length} />
       </form>
-      {error && <div className="error">⚠ {error}</div>}
+      {error && !sending && (
```
Comment (alexfarrill) [inline packages/cli/src/components/PreviewSender.tsx:125]: clear the error message when you send again

File: packages/cli/src/components/PreviewSender.tsx (hunk 1 above)
Comment (psugihara) [reply packages/cli/src/components/PreviewSender.tsx:32]: would this work?

```
  useEffect(() => {
       setEmail(window.localStorage.getItem("previewSenderEmail"));
       const lastSent = window.localStorage.getItem("previewSenderLastSentAt");
       if (lastSent) {
         setLastSentAt(new Date(lastSent));
       }
   }, []);
```

**Review by psugihara — APPROVED** [review]

File: packages/cli/src/components/PreviewSender.tsx (hunk 1 above)
Comment (alexfarrill) [reply packages/cli/src/components/PreviewSender.tsx:32]: yes, but ts lint does not allow

**Review by alexfarrill — COMMENTED** [review]
Nice

**Comment by alexfarrill** [comment]
JSONified error message from the server:

Screenshot 2022-11-22 at 4 22 49 PM

**Pull request merged**